wolfcrypt/src/asn.c

```diff
@@ -5717,7 +5717,7 @@ static int DecodeAltNames(byte* input, int sz, DecodedCert* cert)
             /* Verify RFC 5280 Sec 4.2.1.6 rule:
                 "The name MUST NOT be a relative URI" */
 
-            if (XSTRNCMP((const char*)&input[idx], "://", strLen + 1) != 0) {
+            if (XSTRNSTR((const char*)&input[idx], "://", strLen + 1) == NULL) {
```

Comment:
strcmp starts at the beginning for comparison, strstr will find the needle anywhere in the haystack.  Doesn't an absolute uri have :// at the beginning?

Reply:
The problem encountered is that the URI scheme is not accounted for. For example http://example.com is an absolute URI, and the idx value for input buffer with this comparison was starting at 'h'. My first solution was searching for the first instance of a ':' character encountered but that became similar to just doing a strnstr. I can get it to be more exact, having logic to test that a ':' char is encountered before any '/' char and incrementing an index to skip over scheme's like http, ftp, ldap... then doing the comparision on the first encounter of ':'